cylc/flow/install.py

```diff
@@ -285,13 +285,18 @@ def install_workflow(
     source = Path(expand_path(source)).resolve()
     if not workflow_name:
         workflow_name = get_source_workflow_name(source)
-    validate_workflow_name(workflow_name, check_reserved_names=True)
     if run_name is not None:
         if len(Path(run_name).parts) != 1:
             raise WorkflowFilesError(
                 f'Run name cannot be a path. (You used {run_name})'
             )
+        validate_workflow_name(
+            os.path.join(workflow_name, run_name),
+            check_reserved_names=True
+        )
         check_reserved_dir_names(run_name)
```

Comment:
Isn't the subsequent `check_reserved_dir_names(run_name)` superfluous with `check_reserved_names=True`?

Reply:
Ah yes, good spot (my suggestion https://github.com/cylc/cylc-flow/pull/6264#discussion_r1692729604 removed this but was not applied directly)